src/Symfony/Component/DependencyInjection/Compiler/ServiceReferenceGraph.php

```diff
@@ -80,17 +69,12 @@ public function clear()
 
     /**
      * Connects 2 nodes together in the Graph.
-     *
-     * @param string $sourceId
-     * @param string $sourceValue
-     * @param string $destId
-     * @param string $destValue
-     * @param string $reference
-     * @param bool   $lazy
-     * @param bool   $weak
      */
-    public function connect($sourceId, $sourceValue, $destId, $destValue = null, $reference = null, bool $lazy = false, bool $weak = false)
+    public function connect(?string $sourceId, $sourceValue, ?string $destId, $destValue = null, string $reference = null, bool $lazy = false, bool $weak = false)
     {
+        if (null === $sourceId || null === $destId) {
```

Comment:
shouldn't this be applied in older branches if it is possible to pass `null` in the method ?